app/components/ItunesCard/tests/index.test.js

```diff
@@ -0,0 +1,91 @@
+/**
+ *
+ * Tests for ItunesCard
+ *
+ */
+
+import { renderWithIntl } from '@utils/testUtils';
+import { fireEvent } from '@testing-library/dom';
+
+import React from 'react';
+import ItunesCard from '../index';
+import { itune } from './mockData';
+import { truncate } from 'lodash';
+
+describe('<ItunesCard />', () => {
+  let handleOnActionClick;
+
+  beforeEach(() => {
+    itune, (handleOnActionClick = jest.fn());
+  });
+  it('should render and match the snapshot', () => {
+    const { baseElement } = renderWithIntl(<ItunesCard itune={itune} handleOnActionClick={handleOnActionClick} />);
+    expect(baseElement).toMatchSnapshot();
+  });
+
+  it('should contain 1 ItunesCard component', () => {
+    const { getAllByTestId } = renderWithIntl(<ItunesCard itune={itune} handleOnActionClick={handleOnActionClick} />);
+    expect(getAllByTestId('itune-card').length).toBe(1);
+  });
+
+  it('should check if the song details are rendered inside the card and progress bar should be in the card', () => {
+    const { getByTestId } = renderWithIntl(<ItunesCard itune={itune} handleOnActionClick={handleOnActionClick} />);
+
+    expect(getByTestId('itune-card')).toHaveTextContent(itune.trackName);
```

Comment:
we need 2 tests here. If these values are empty in the track object, then we should test that they are not rendered

Reply:
added the test